lib/diversity/component.rb

```diff
@@ -56,30 +56,26 @@ def initialize(spec, options)
       @options = DEFAULT_OPTIONS.keep_merge(options)
       @logger = @options[:logger]
 
-      schema = JsonSchemaCache[
-                 MASTER_COMPONENT_SCHEMA,
-                 { validate_spec: @options[:validate_spec] }
-               ]
-
       # Handle both Hash and String specs
       @raw = (spec.is_a?(String) ? parse_config(spec) : spec)
 
-      validation = schema.validate(@raw)
-      raise Diversity::Exception, "Configuration is not valid:\n" + validation.join("\n"), caller if
-        validation.length > 0
+      if (@options[:validate_spec])
+        schema = JsonSchemaCache[
+          MASTER_COMPONENT_SCHEMA,
+          { validate_spec: @options[:validate_spec] }
+        ]
+
+        validation = schema.validate(@raw)
+        raise Diversity::Exception,
+              "Configuration is not valid:\n" + validation.join("\n"), caller if
+          validation.length > 0
+      end
 
-      @checksum = Digest::SHA1.hexdigest(dump)
+      @checksum = "#{name}:#{version}"
       @assets = {}
       populate(@raw)
     end
 
-    # Returns a JSON dump of the component configuration
-    #
-    # @return [String]
-    def dump
```

Comment:
I removed it since it wasn't used.  This code is so full of things that someone MIGHT need some day that it takes 6 seconds to render a simple page that nodejs on a weaker machine renders on 1 second.
We might need a serialize-method of some sort; we'll make one when we need it.  If the gotten diversity.json is cached by the Registry, this dump does not add anything.